### ISSUE TITLE
Bad EKF results on pegasus dataset

### ISSUE THREAD

**Issue by maulanaazhari** [Air-IO/Air-IO#1]
Hi! Thanks for open-sourcing the code!

I tried to run the trained model on pegasus dataset. However, I found out that the air-io with ekf is worse than the just air-io.

This is the result that I got:
### TEST_1

EKF result:
![Image](https://github.com/user-attachments/assets/f710ce33-7fd2-4ab9-a286-5dda9047a0e3)
bias:
![Image](https://github.com/user-attachments/assets/2b35e75e-0400-41c5-8c68-bbfb618bc5e5)
rotation:
![Image](https://github.com/user-attachments/assets/bc984acd-560a-479a-b58e-acf6fdc2c89b)
velocity:
![Image](https://github.com/user-attachments/assets/07e23da1-c687-4ba3-ba67-4a386791e73f)

### TEST_2

EKF result:
![Image](https://github.com/user-attachments/assets/e0df6e97-7d6a-4eee-9445-e529f7e493cb)
bias:
![Image](https://github.com/user-attachments/assets/221773c1-406e-4341-ae88-db4cd4c8adf4)
rotation:
![Image](https://github.com/user-attachments/assets/ecb85b04-2c31-4193-8f97-4bf15905e8d1)
vel:
![Image](https://github.com/user-attachments/assets/82c77d12-3ad5-4585-88f8-7c72034f7921)

I realized that the EKF bias is always zero, is it related to the results?
Thanks

**Comment by haleqiu** [comment]
@maulanaazhari Thank you for pointing this out. I think the bias is due to some bug in the pypose EKF we wrote. We were using pytorch auto-grad to linearize the system. Some gradients are broken. The fix will be upload soon. Let's keep in touch.

**Comment by wangqin666** [comment]
May I ask if this issue has been resolved

**Comment by haleqiu** [comment]
@maulanaazhari  Hi, thank you for bringing up this. We updated the code. Please let us know if you have other issues.

**Comment by sleepycan** [comment]
Hello, @maulanaazhari , additionally, regarding the Pegasus dataset, could you please also download and try the updated [AirIMU_pegasus](https://github.com/Air-IO/Air-IO/releases/tag/AirIMU)?  Let us know if you have other issues.
